src/condor_tools/history.cpp

```diff
@@ -908,33 +908,22 @@ static void readHistoryFromFiles(const char* matchFileName, const char* constrai
 		exit(1);
 	}
 
-        // The user didn't specify the name of the file to read, so we read
-        // the history file, and any backups (rotated versions). 
-        int numHistoryFiles;
-        const char **historyFiles;
+	// Find all time rotated files matching the filename in its given directory
+	std::vector<std::string> historyFiles = findHistoryFiles(matchFileName);
```

Comment:
this idiom means that you can't add files into an existing collection, better to pass the vector in as a reference argument

Reply:
Decided to leave this a return idiom and a vector. If we do decide to allow adding to an already made data structure then this can change.